### PR TITLE
fix: add apply strategy in work

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

e2e tests

### CODE REVIEW

File: test/e2e/placement_apply_strategy_test.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+				},
+			})
+			ns.Annotations = map[string]string{
+				annotationKey: annotationValue,
+			}
+			By(fmt.Sprintf("creating namespace %s on member cluster", ns.Name))
+			Expect(memberCluster1EastProdClient.Create(ctx, &ns)).Should(Succeed(), "Failed to create namespace %s", ns.Name)
+
+			// Create the CRP.
+			strategy := &placementv1beta1.ApplyStrategy{AllowCoOwnership: true}
```
Comment (ryanzhang-oss) [inline test/e2e/placement_apply_strategy_test.go:59]: can we also test the case it's not filled?

File: test/e2e/placement_apply_strategy_test.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		// This check will ignore the annotation of resources.
+		It("should place the selected resources on member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+		It("should have annotations on the namespace", func() {
+			want := map[string]string{annotationKey: annotationValue}
+			Expect(validateAnnotationOfWorkNamespaceOnCluster(memberCluster1EastProd, want)).Should(Succeed(), "Failed to override the annotation of work namespace on %s", memberCluster1EastProdName)
+		})
+
+		It("can delete the CRP", func() {
+			// Delete the CRP.
```
Comment (ryanzhang-oss) [inline test/e2e/placement_apply_strategy_test.go:82]: we delete crp in the afterAll block

File: test/e2e/placement_apply_strategy_test.go (hunk 1 above)
Comment (zhiying-lin) [reply test/e2e/placement_apply_strategy_test.go:59]: we do have one test in https://github.com/Azure/fleet/blob/main/test/e2e/placement_selecting_resources_test.go#L691

File: test/e2e/placement_apply_strategy_test.go (hunk 2 above)
Comment (zhiying-lin) [reply test/e2e/placement_apply_strategy_test.go:82]: each test will delete the crp in the afterAll block in line 65 to make sure we have a clean test env.

File: test/e2e/setup.sh [hunk 3]
```diff
@@ -118,6 +118,7 @@ helm install hub-agent ../../charts/hub-agent/ \
     --set image.repository=$REGISTRY/$HUB_AGENT_IMAGE \
     --set image.tag=$TAG \
     --set namespace=fleet-system \
+    --set logVerbosity=5 \
```
[the remark this answers is not in the document]
Comment (zhiying-lin) [reply test/e2e/setup.sh:121]: i thought the default value is 2.
it's ok to set it explicitly in case we change the default value.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**